lsp-mode.el

```diff
@@ -5939,6 +5942,27 @@ SESSION is the active session."
   :type 'directory
   :package-version '(lsp-mode . "6.3"))
 
+(defcustom lsp-server-locations nil
```

Comment:
Also what's the advantage of adding a new custom variable instead of just calling `lsp-dependency` with new values for `npm` backend?
User doesn't need one more customizable variable and we don't need to change big chunks of code.

In that case you can probably make `lsp-dependency` autoload.

Reply:
It would be nice to have a way to eliminate the customize I proposed. I couldn't see how to inject the location of the language server without adding a defcustom. How would I go about doing this with a call to lsp-dependency? Alternatively, if you'd like I can try making the defcusom more general to handle all types of language servers, though would need some pointers.

In our environment we configure Emacs for a large number of programmers. They get all of the benefits of lsp, etc. without having to set anything up. The add-on packages are placed on NFS / Samba for Linux, Mac, Windows users. This is a significant time saver, especially when many people are not comfortable with lisp.

Reply:
I see, taking a closer look at the code again, how about changing `lsp-server-install-dir` to the shared storage path?
You can probably put that as local-dirs variable inside project folder so everyone who clone the project can benefit from that.